src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -628,61 +632,23 @@ bool IsChangeAddress(Script scriptPubKey)
             return model;
         }
 
-        private GetTransactionModel GetWatchOnlyTransaction(uint256 trxid)
+
+        /// <summary>
+        /// We get the details via the wallet service's history method.
+        /// </summary>
+        private async Task<WalletHistoryModel> GetWatchOnlyTransactionAsync(uint256 trxid)
         {
             var accountReference = this.GetWatchOnlyWalletAccountReference();
 
-            var hdAccount = this.walletManager.GetOrCreateWatchOnlyAccount(accountReference.WalletName);
-
-            var history = this.walletManager.GetHistory(hdAccount);
-            if (history == null)
-                throw new RPCServerException(RPCErrorCode.RPC_INVALID_ADDRESS_OR_KEY, "Watch only account not found on the wallet.");
-
-            var transactionData = history.History.FirstOrDefault(t => t.Transaction.Id == trxid);
-            if (transactionData == null)
-                throw new RPCServerException(RPCErrorCode.RPC_INVALID_ADDRESS_OR_KEY, "Invalid or non-wallet transaction id.");
-
-            var transactionModel = new GetTransactionModel
+            var request = new WalletHistoryRequest()
             {
-                Confirmations = transactionData.Transaction.BlockHeight != null ? this.ConsensusManager.Tip.Height - transactionData.Transaction.BlockHeight.Value + 1 : 0,
-                BlockHash = transactionData.Transaction.BlockHash,
-                BlockIndex = transactionData.Transaction.BlockIndex,
-                Isgenerated = (transactionData.Transaction.IsCoinBase ?? false) || (transactionData.Transaction.IsCoinStake ?? false),
-                TransactionId = trxid,
-                TransactionTime = transactionData.Transaction.CreationTime.ToUnixTimeSeconds(),
-                TimeReceived = transactionData.Transaction.CreationTime.ToUnixTimeSeconds(),
-                Details = new List<GetTransactionDetailsModel>(),
-                Hex = transactionData.Transaction.Hex
+                WalletName = accountReference.WalletName,
+                AccountName = "watchOnly",
```

Comment:
Maybe this should be the const from `Wallet`

Reply:
Done 👍